### PR TITLE
Limit number of user addresses

### PR DESCRIPTION
**Pull request by IKarbowiak**
- add migration to left only 100 addresses per user
- introduce the settings that defined the maximum amount of addresses per user
- do not save a new address in the checkout if the user already has max amount of addresses
- when a new address to the user with a max number of addresses, remove the older one

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-5936 Users with a lot of addresses can load overload Saleor.](https://app.clickup.com/t/2549495/SALEOR-5936)

File: saleor/account/utils.py [hunk 1]
```diff
@@ -16,6 +18,9 @@ def store_user_address(
     manager: "PluginsManager",
 ):
     """Add address to user address book and set as default one."""
+    if is_user_address_limit_reached(user):
```
Comment (fowczarek) [inline saleor/account/utils.py:21]: Could we add some docstring that describes this behavior?

File: saleor/account/migrations/0057_clear_user_addresses.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+from django.db import migrations
+from django.db.models import Count
+
+
+def clear_addresses(apps, schema_editor):
+    User = apps.get_model("account", "User")
+    Address = apps.get_model("account", "Address")
+
+    users = User.objects.annotate(address_count=Count("addresses")).filter(
+        address_count__gt=100
```
Comment (tomaszszymanski129) [inline saleor/account/migrations/0057_clear_user_addresses.py:12]: we use 100 here and below, maybe worth extracting it to meaningfull const like `USER_ADDRESS_LIMIT`

File: saleor/account/utils.py [hunk 3]
```diff
@@ -31,6 +36,31 @@ def store_user_address(
...
+def remove_the_oldest_user_address(user: "User"):
+    user_default_addresses_ids = [
+        user.default_billing_address_id,
+        user.default_shipping_address_id,
+    ]
+    user_address = (
+        user.addresses.exclude(pk__in=user_default_addresses_ids).order_by("pk").first()
+    )
+    if user_address:
+        user_address.delete()
```
Comment (tomaszszymanski129) [inline saleor/account/utils.py:61]:
```suggestion
    user_address := (
        user.addresses.exclude(pk__in=user_default_addresses_ids).order_by("pk").first()
    ):
        user_address.delete()
```

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/account/utils.py (hunk 3 above)
Comment (IKarbowiak) [reply saleor/account/utils.py:63]: I'm not sure if it's more readable than the previous solution.

File: saleor/account/migrations/0057_clear_user_addresses.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Generated by Django 3.2.12 on 2022-02-15 14:18
+
+from django.db import migrations
+from django.db.models import Count
+
+USER_ADDRESS_LIMIT = 100
```
Comment (L3str4nge) [inline saleor/account/migrations/0057_clear_user_addresses.py:6]: Shouldn't be `settings.MAX_USER_ADDRESSES = int(os.environ.get("MAX_USER_ADDRESSES", 100))` ?

**Review by L3str4nge — APPROVED** [review]

**Review by fowczarek — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Pull request merged**